apps/explorer/src/hooks/useGetOwnedObjects.ts

```diff
@@ -3,21 +3,33 @@
 
 import { useRpcClient } from '@mysten/core';
 import { type SuiAddress } from '@mysten/sui.js';
-import { useQuery } from '@tanstack/react-query';
+import { useInfiniteQuery } from '@tanstack/react-query';
+
+const MAX_OBJECTS_PER_REQ = 6;
 
 export function useGetOwnedObjects(address?: SuiAddress | null) {
     const rpc = useRpcClient();
-    return useQuery(
+    return useInfiniteQuery(
         ['get-owned-objects', address],
-        async () =>
+        async ({ pageParam }) =>
             await rpc.getOwnedObjects({
                 owner: address!,
                 options: {
                     showType: true,
                     showContent: true,
                     showDisplay: true,
                 },
+                limit: MAX_OBJECTS_PER_REQ,
+                cursor: pageParam ? pageParam.cursor : null,
             }),
-        { enabled: !!address }
+        {
+            enabled: !!address,
+            getNextPageParam: (lastPage) =>
+                lastPage?.hasNextPage
+                    ? {
```

Comment:
Just curious why not just pass `lastPage.nextCursor` directly?

Reply:
So the way the `hasNextPage` is emitted from this hook is based on what `getNextPageParam` evaluates to - if you emit the lastCursor of the last page, you will tell the component using the hook that there is a next page to fetch when there in fact isn't. I think the API could be updated to return null for a lastCursor if you've fetched the last page, but I'm not sure what the impact would be there.